deploy/AllocatorInit.sol

```diff
@@ -0,0 +1,295 @@
+
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, ilkInstance.swapper,        SwapperLike.swap.selector,            true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, ilkInstance.depositorUniV3, DepositorUniV3Like.deposit.selector,  true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, ilkInstance.depositorUniV3, DepositorUniV3Like.withdraw.selector, true);
+        RolesLike(sharedInstance.roles).setRoleAction(ilk, cfg.automationRole, ilkInstance.depositorUniV3, DepositorUniV3Like.collect.selector,  true);
+
+        // Set the allocator proxy as the ilk admin instead of the Pause Proxy
+        RolesLike(sharedInstance.roles).setIlkAdmin(ilk, cfg.allocatorProxy);
+
+        // Allow facilitator to set configurations in the automation contracts
+        for(uint256 i = 0; i < cfg.facilitators.length; i++) {
+            WardsLike(ilkInstance.stableSwapper).rely(cfg.facilitators[i]);
+            WardsLike(ilkInstance.stableDepositorUniV3).rely(cfg.facilitators[i]);
+            WardsLike(ilkInstance.conduitMover).rely(cfg.facilitators[i]);
+        }
+
+        // Add keepers to the automation contracts
+        for(uint256 i = 0; i < cfg.stableSwapperKeepers.length; i++) {
+            KissLike(ilkInstance.stableSwapper).kiss(cfg.stableSwapperKeepers[i]);
+        }
+        for(uint256 i = 0; i < cfg.stableDepositorUniV3Keepers.length; i++) {
+            KissLike(ilkInstance.stableDepositorUniV3).kiss(cfg.stableDepositorUniV3Keepers[i]);
+        }
+        for(uint256 i = 0; i < cfg.conduitMoverKeepers.length; i++) {
+            KissLike(ilkInstance.conduitMover).kiss(cfg.conduitMoverKeepers[i]);
+        }
+
+        // Move ownership of the ilk contracts to the allocator proxy
+        switchOwner(ilkInstance.vault,                ilkInstance.owner, cfg.allocatorProxy);
+        switchOwner(ilkInstance.buffer,               ilkInstance.owner, cfg.allocatorProxy);
+        switchOwner(ilkInstance.swapper,              ilkInstance.owner, cfg.allocatorProxy);
+        switchOwner(ilkInstance.depositorUniV3,       ilkInstance.owner, cfg.allocatorProxy);
+        switchOwner(ilkInstance.stableSwapper,        ilkInstance.owner, cfg.allocatorProxy);
+        switchOwner(ilkInstance.stableDepositorUniV3, ilkInstance.owner, cfg.allocatorProxy);
+        switchOwner(ilkInstance.conduitMover,         ilkInstance.owner, cfg.allocatorProxy);
+
+        // Add allocator-specific contracts to changelog
+        string memory ilkString = ScriptTools.ilkToChainlogFormat(ilk);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(ilkString, "_ALLOCATOR_VAULT"))),  ilkInstance.vault);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(ilkString, "_ALLOCATOR_BUFFER"))), ilkInstance.buffer);
```

Comment:
Hmm in general we add the `ilk` as a suffix. We should probably also consider the case the idea the `ilk` might come with a "-", so that needs to be converted to "_"

Reply:
I think that's correct for the keys with `MCD` or `PIP` prefix, but not for others like `RWA003_A_INPUT_CONDUIT` or `DIRECT_COMPV2_DAI_POOL`.

The conversion from "-" to "_" is done already in `ilkToChainlogFormat`.
I did push a change now to set the ilk registry name and symbol with "-".

Reply:
Ok yeah you are right, forgot that function converted the dash to underscore.
For the key I'm just wondering if it would be better to be able to group everything related to the `ALLOCATOR` ecosystem with that prefix.
I would even argue we could put the `ilk` in the middle so you have:
```
ALLOCATOR_ROLES
ALLOCATOR_REGISTRY
ALLOCATOR_ORACLE
ALLOCATOR_XXX_VAULT
ALLOCATOR_XXX_BUFFER
ALLOCATOR_YYY_VAULT
ALLOCATOR_YYY_BUFFER
```

Btw I'm still not 100% convinced the oracle key should not be under the subdao key name, even thinking that all of them will have the same. Just wondering a case where we could need a different oracle for an `ilk` (due to some random reason), then we will need to start messing up with the keys.

Reply:
1. The discussion about the CL key really depends on what the ilk itself would look like (which tbh I didn't want to have now and that's why I didn't set the key automatically initially).
If we need to guess I'd assume it would be something like `ALLOCATOR001` / `AD001` / `ADAO001`.

So first of all, I think you would be able to group all of the keys together anyhow, even if the ilk is a prefix (as you can currently for ilks like  `DIRECT_COMPV2_DAI` and `DIRECT-AAVEV2-DAI`, or `RWA001` and `RWA002`).

I'm thinking we can probably even remove the "ALLOCATOR" from the keys and just have something like `ALLOCATOR001_VAULT`  / `AD001_VAULT` / `ADAO001_VAULT`.

2. Not sure about the oracle key/keys. I don't really see such duplication in the current chainlog values.

Reply:
Hmmm not sure about removing ALLOCATOR keyword yet, I was expecting to use real names for the ´ilks´ as "SPARK" but the naming is coming later, right?

Regarding the oracles, hmmm `PIP_WBTC` and `PIP_RENBTC` point to the same contract for example.

Reply:
I think most chances the ilk will not just be something like "SPARK" but will need some indicator of the type of ilk it is (that's at least what's done so far for d3ms and rwas).

Regarding the oracles, you are right about that the BTC pips. I changed it to have a pip entry per ilk.
Currently the "PIP_" is a prefix while the contract type is a suffix (same as rwas).